### PR TITLE
Styling files

### PR DESCRIPTION
**Pull request by VicenteCartas**
Adding editorconfig and Intellij default settings following MegaMek Code Style Guide.

Read more about using this on Intellij here: https://www.jetbrains.com/help/idea/configuring-code-style.html

### CODE REVIEW

File: Intellij-Java-Default.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<code_scheme name="Default" version="173">
+  <codeStyleSettings language="JAVA">
+    <option name="KEEP_LINE_BREAKS" value="false" />
+    <option name="KEEP_FIRST_COLUMN_COMMENT" value="false" />
+    <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
+    <option name="IF_BRACE_FORCE" value="3" />
```
Comment (VicenteCartas) [inline Intellij-Java-Default.xml]: These settings indicate force braces even if the expression is a single line.

File: MekHQ/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# EditorConfig: https://EditorConfig.org
+
+[*]
+end_of_line = crlf
```
Comment (sixlettervariables) [inline MekHQ/.editorconfig]: Three things:
1. I don't know that the project uses `crlf`. I know at least a few us use non-Windows. git should be controlling this for us anyways with core.crlf auto mode. I'd wager these files are actually `lf`.
2. Add `charset = utf-8`
3. Add `insert_final_newline = true`

File: MekHQ/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+[*.java]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
```
Comment (sixlettervariables) [inline MekHQ/.editorconfig]: I suggest adding:
1. `continuation_indent_size = 8`

Also:
```editorconfig
[*.xml]
indent_style = space
indent_size = 4
trim_trailing_whitespace = true
```

File: MekHQ/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+insert_final_newline = true
+continuation_indent_size = 8
+
+[*.xml]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
```
Comment (VicenteCartas) [inline MekHQ/.editorconfig]: Addressing PR comments.

File: MekHQ/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# EditorConfig: https://EditorConfig.org
+
+[*]
+charset = utf-8
+
+[*.java]
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (sixlettervariables) [inline MekHQ/.editorconfig]: `insert_final_newline` and `trim_trailing_whitespace` should ideally be in `[*]` rather than in `[*.java]` and `[*.xml]`.

Comment (VicenteCartas) [reply MekHQ/.editorconfig]: Good point, addressed!

**Pull request merged**